src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -95,84 +103,69 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
             if (method.IsVoidReturningAsync())
             {
                 return TryCreate(
-                    F: F,
-
+                    F,
                     builderType: F.WellKnownType(WellKnownType.System_Runtime_CompilerServices_AsyncVoidMethodBuilder),
                     resultType: F.SpecialType(SpecialType.System_Void),
-
+                    createBuilderMethod: null,
+                    createBuilder: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__Create,
                     setException: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__SetException,
                     setResult: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__SetResult,
                     awaitOnCompleted: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__AwaitOnCompleted,
                     awaitUnsafeOnCompleted: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__AwaitUnsafeOnCompleted,
                     start: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__Start_T,
                     setStateMachine: WellKnownMember.System_Runtime_CompilerServices_AsyncVoidMethodBuilder__SetStateMachine,
-                    task: null,
+                    task: WellKnownMember.Count,
```

Comment:
Deserves a comment if you're using this as a sentinel! I wonder if you might maybe have a constant with this value but a more descriptive name?